app/components/view-goal/line-chart-detail.jsx

```diff
@@ -4,65 +4,109 @@ import moment from 'moment';
 import GoalTileComponent from '../goal-tile.jsx';
 import LineChart from './line-chart.jsx';
 
-const LineChartDetail = ({ currentGoal, onSelectRating, dynamicStyle, }) => {
-
-  const allRatings = currentGoal.ratings.slice(0).reverse();
-
-  const containerStyle = {
-    width: allRatings.length < 6
-      ? '100%'
-      : 400 + (allRatings.length - 3) * 10,
-  };
-
-  const feedbackStyle = {
-    opacity: currentGoal.ratingSelected
-      ? 1
-      : 0,
-  };
-
-  const time = currentGoal.ratingSelected
-    ? currentGoal.ratingSelected.time
-    : {};
-
-  const score = currentGoal.ratingSelected
-    ? currentGoal.ratingSelected.score
-    : null;
-
-  const comment = currentGoal.ratingSelected
-    ? currentGoal.ratingSelected.comment
-    : '';
-
-  return (
-   <div className="line-chart-detail goal-detail-page" style={ dynamicStyle }>
-     <div className="goal-detail-goal-tile-container">
-       <GoalTileComponent goal={ currentGoal } />
-     </div>
-     <div className="line-chart-container-detail">
-       <div className="line-chart-detail-feedback-container" style={ feedbackStyle }>
-         <p className="detail-feedback-time">
-           { moment(time).format('LLLL') }
-         </p>
-         <p className="detail-feedback-score">
-           You rated your progress { score }/10
-         </p>
-         <p className="detail-feedback-comment">
-           { comment }
-         </p>
+const defaultWidth = 400;
+const animationName = 'scroll';
+
+class LineChartDetail extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      customWidth: props.currentGoal.ratings.length < 6
```

Comment:
Does this change? It does not look like it does..
If it does, maybe think about moving it into redux.

If it does not change, it should not be in the state. Either you can move this functionality into mapStateToProps or just make a helper here..

Might be easier if you told me why it is here..